Lib/unittest/mock.py

```diff
@@ -2482,6 +2484,55 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event = event_class()
+        self._expected_calls = {}
+
+    def _mock_call(self, *args, **kwargs):
+        ret_value = _safe_super(WaitableMock, self)._mock_call(*args, **kwargs)
+
+        for call in self._mock_mock_calls:
+            event = self._expected_calls.get(call.args)
+            if event and not event.is_set():
+                event.set()
+
+        self._event.set()
+
+        return ret_value
+
+    def wait_until_called(self, timeout=1.0):
+        """Wait until the mock object is called.
+
+        `timeout` - time to wait for in seconds. Defaults to 1.
+        """
+        return self._event.wait(timeout=timeout)
+
+    def wait_until_called_with(self, *args, timeout=1.0):
+        """Wait until the mock object is called with given args.
+        If args is empty then it waits for the mock object to be called.
+
+        `timeout` - time to wait for in seconds. Defaults to 1.
```

Comment:
I don’t have a great alternative, but feels a pity if users cannot wait for calls that had a timeout parameter as discussed in the issue.

Reply:
Yes, I would wait if someone has a less conflicting name or perhaps receive `kwargs_dict` as a dictionary instead of `(**kwargs, timeout)` that fixes conflict but the API would be nice to pass keyword arguments as **kwargs itself.